src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -123,13 +126,39 @@ public Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedR
         return m_photonPoseEstimator.update();
     }
 
-    // private Optional<Pose2d> getTagPose(int tagID){
-    //     // This can fail if the tagID is not part of the map. 
-    //     // Need to check before converting to Pose2d
-    //     return Optional.of(m_aprilTagFieldLayout.getTagPose(tagID).get().toPose2d());
-    // }
+    public Optional<Pose2d> getCentralTagPose() { // gets target closets to center of camera
+        var targetResult = m_aprilTagCamera.getLatestResult();
+        int targetID = -1;
+        if (targetResult.hasTargets()) {
+            // Get lists of targets
+            List<PhotonTrackedTarget> targets = targetResult.getTargets();
+            // make a temp holder var for least Y translation, set to first tags translation
+            //Translation2d translationHolder = targets.get(0).getBestCameraToTarget().getTranslation().toTranslation2d();
+            double minTranslation = targets.get(0).getBestCameraToTarget().getTranslation().toTranslation2d().getY();
+            for (PhotonTrackedTarget tag : targets) { // for every target in camera
+                // get transformation to target
+                Transform3d tagTransform = tag.getBestCameraToTarget();
+                // get translation to target from transformation
+                Translation2d tagTranslation = tagTransform.getTranslation().toTranslation2d();
+
+                // looking for smallest absolute relative to camera Y
+                // if abs Y translation of new tag is less then holder tag, it becomes holder tag
+                if (Math.abs(tagTranslation.getY()) < Math.abs(minTranslation)) {
+                    minTranslation = tagTranslation.getY();
+                    targetID = tag.getFiducialId(); // set targetID
+                }
+            }
+        }
+        //optional in case no target is found
+        return Optional.of(m_aprilTagFieldLayout.getTagPose(targetID).get().toPose2d());
+
+    }
+
+    public Optional<Pose2d> getTagPose(int tagID) {
+        return Optional.of(m_aprilTagFieldLayout.getTagPose(tagID).get().toPose2d());
```

Comment:
This is wrong too. See the comment in the version in branch main. Please just delete this routine.